test_runner/src/main/kotlin/ftl/args/AndroidArgs.kt

```diff
@@ -91,24 +91,27 @@ AndroidArgs
     }
 }
 
+// Changes these based on type
 val AndroidArgs.isDontAutograntPermissions
     get() = !(grantPermissions.isNotNull() && (grantPermissions.equals("all")))
 
 val AndroidArgs.isInstrumentationTest
-    get() = appApk.isNotNull() &&
-            testApk.isNotNull() ||
-            additionalAppTestApks.isNotEmpty() &&
-            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() })
+    get() = if (type != null) (type == Type.INSTRUMENTATION) else
+        (appApk.isNotNull() && testApk.isNotNull() || additionalAppTestApks.isNotEmpty() &&
+            (appApk.isNotNull() || additionalAppTestApks.all { (app, _) -> app.isNotNull() }))
 
 val AndroidArgs.isRoboTest
-    get() = appApk.isNotNull() &&
-            (roboDirectives.isNotEmpty() || roboScript.isNotNull())
+    get() = if (type != null) (type == Type.ROBO) else
+        (appApk.isNotNull() && (roboDirectives.isNotEmpty() || roboScript.isNotNull()))
```

Comment:
this is better, but also could be wrap in `validateRobo` function

Reply:
👍